### ISSUE TITLE
Guidelines around user-defined disaggregated metrics

### ISSUE THREAD

**Issue by michaelamoako** [fairlearn/fairlearn#786]
#### Describe the issue linked to the documentation
As Fairlearn and the associated Fairness dashboard look to expand beyond y_true, y_pred metric formulations (which is geared towards Tabular data), there may need to be documentation around how to think about disaggregated metrics in general with some examples of metrics for unstructured data scenarios that can be disaggregated. 

**Related issues**

Fairlearn: https://github.com/fairlearn/fairlearn/issues/756

RAI Widgets: https://github.com/microsoft/responsible-ai-widgets/issues/498

#### Suggest a potential alternative/fix
Documentation that provides some examples and guidelines around choosing user-defined disaggregated metrics, with some examples (both of custom metrics that can be disaggregated, and ones that can not be).

**Comment by riedgar-ms** [comment]
I agree that we need to expand beyond the tabular data outlook of `MetricFrame` - this discussion has arisen in #756 .

I do wonder whether we should be figuring out what new API would be appropriate, rather than trying to extend `MetricFrame`, though. The latter does its job relatively well, and I wouldn't want that compromised.

**Comment by Anish8800** [comment]
Thanks for raising this — I agree that supporting disaggregated metrics beyond tabular y_true / y_pred formulations will be increasingly important as Fairlearn is used with NLP, vision, and multimodal systems.

I also think the point about not overloading MetricFrame is valid. A separate API or documentation section focused on “disaggregated evaluation for unstructured data” could prevent forcing everything into the tabular mental model.

Some examples of disaggregated metrics for unstructured data that might be useful in documentation:

NLP
• toxicity false positive rate by demographic group
• coreference resolution accuracy by gender/race mentions
• translation BLEU/COMET scores disaggregated by language or dialect

Computer vision
• face recognition error rates disaggregated by skin tone, age group, gender presentation
• object detection miss rates across geographic or socioeconomic contexts

Speech
• word error rate across accents or language varieties

It might also help if the documentation distinguishes:
• metrics that naturally disaggregate (group-conditioned)
• metrics where disaggregation is not meaningful or breaks assumptions

Happy to contribute to drafting such documentation or examples if helpful.
